src/tools/schema2cpp.ts

```diff
@@ -57,6 +57,24 @@ export class Schema2Cpp extends Schema2Base {
     return baseName.toLowerCase().replace(/\.arcs$/, '').replace(/[._]/g, '-') + '.h';
   }
 
+  generateEntityClassName(node: SchemaNode, i: number) {
+    if (i === -1) {
+      return `${node.particleName}_${node.connections[0]}`;
+    }
+    return `${node.particleName}Internal${i}`;
+  }
+
+  generateAliasNames(node: SchemaNode): string[] {
+    const arr: string[] = [];
+    if (node.connections.length === 1) {
+      return arr;
```

Comment:
Optional: I'd make this `return [];` and then move the if-statement above the declaration of `arr`. That makes it easier to see what this is actually returning, and keeps the declaration of `arr` closer to where it's used.